tools/cibadmin.c

```diff
@@ -337,6 +326,66 @@ cibadmin_pre_query(int *call_options, xmlNode *input, gchar **acl_user,
     return CRM_EX_OK;
 }
 
+static int
+print_xml_id(xmlNode *xml, void *user_data)
+{
+    const char *id = pcmk__xe_id(xml);
+
+    if (id != NULL) {
+        printf("%s\n", id);
+    }
+    return pcmk_rc_ok;
+}
+
+static void
+cibadmin_output_xml(xmlNode *output, int call_options, const gchar *acl_user,
+                    crm_exit_t *exit_code, GError **error)
+{
+    if ((options.acl_render_mode != pcmk__acl_render_none)
+        && (*exit_code == CRM_EX_OK)
+        && pcmk__xe_is(output, PCMK_XE_CIB)) {
+
+        xmlDoc *acl_evaled_doc = NULL;
+        xmlChar *rendered = NULL;
+        int rc = pcmk__acl_annotate_permissions(acl_user, output->doc,
+                                                &acl_evaled_doc);
+
+        if (rc != pcmk_rc_ok) {
+            *exit_code = CRM_EX_CONFIG;
+            g_set_error(error, PCMK__EXITC_ERROR, *exit_code,
+                        "Could not evaluate access per request (%s, error: %s)",
+                        acl_user, pcmk_rc_str(rc));
+            return;
+        }
+
+        rc = pcmk__acl_evaled_render(acl_evaled_doc, options.acl_render_mode,
+                                     &rendered);
+        if (rc != pcmk_rc_ok) {
+            *exit_code = CRM_EX_CONFIG;
+            g_set_error(error, PCMK__EXITC_ERROR, *exit_code,
+                        "Could not render evaluated access: %s",
```

Comment:
While I'm being picky, I don't like this error message or the one above it either.

Reply:
added a commit on top